test/preprocess.spec.js

```diff
@@ -1,90 +1,152 @@
 /* global describe it */
+
 // @ts-check
+// The previous line must be the first non-comment line in the file to enable TypeScript checks:
+// https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check
+
 const fs = require('fs')
 const chai = require('chai')
 const expect = chai.expect
 const dirtyChai = require('dirty-chai')
 const path = require('path')
+const url = require('url')
 
 chai.use(dirtyChai)
 
-const { readFixture } = require('./utils.js')
 const { preprocessVegaLite } = require('../src/preprocess.js')
 
 describe('Vega-Lite preprocessing', () => {
+  const cwd = process.cwd().replace(/\\/g, '/')
+  const relativePath = 'test/fixtures/vegalite-data.csv'
+  const diagramTextWithInlinedCsvFile = JSON.stringify({
+    data: {
+      values: fs.readFileSync(`${path.join(cwd, relativePath)}`, 'utf8'),
+      format: {
+        type: 'csv'
+      }
+    }
+  })
+
+  /**
+   * @param {string} diagramText
+   * @param {string} expectedErrorMessage
+   * @param {string} [baseDir]
+   * @returns {void}
+   */
+  function expectToThrow (diagramText, expectedErrorMessage, baseDir) {
+    expect(() => preprocessVegaLite(diagramText, {}, baseDir)).to.throw(expectedErrorMessage)
+  }
+
+  /**
+   * @param {string} diagramText
+   * @param {string} expectedPreprocessedDiagramText
+   * @param {string} [baseDir]
+   * @returns {void}
+   */
+  function expectToBeEqual (diagramText, expectedPreprocessedDiagramText, baseDir) {
+    expect(preprocessVegaLite(diagramText, {}, baseDir)).to.be.equal(expectedPreprocessedDiagramText.replace(/\r\n/g, '\n'))
+  }
```

Comment:
Would it be possible to revert these changes? It would be easier to review :angel: 
We can still introduce them later.

Reply:
Yes, I agree that it is very difficult to compare with the previous version. But even if I inline these helper functions, the diff would probably not be very helpful because the other changes would be too big. For example, I changed the test names, added and moved tests, extracted constants....
Could you please try to review without comparing with the previous version? If the test cases are not comprehensible, I am happy to improve them. 
Sometimes it is difficult to me to stop refactoring code ...

Reply:
> Sometimes it is difficult to me to stop refactoring code ...

I do have the same "problem" 😃

> Could you please try to review without comparing with the previous version? If the test cases are not comprehensible, I am happy to improve them.

Fair enough, I will checkout the branch locally and let you know 👌